mono/metadata/mono-config.c

```diff
@@ -642,9 +642,10 @@ mono_config_for_assembly_internal (MonoImage *assembly)
 	g_free (cfg_name);
 
 	cfg_name = g_strdup_printf ("%s.config", mono_image_get_name (assembly));
+	const char *mono_cfg_dir = mono_get_config_dir ();
 
-	for (i = 0; (aname = get_assembly_filename (assembly, i)) != NULL; ++i) {
```

Comment:
Just check before https://github.com/mono/mono/pull/12056/files#diff-7f8a7c5a525275537e8961a9c56f8568R644, and do an early exit.

Reply:
Addressed